codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/ManifestWriter.java

```diff
@@ -0,0 +1,157 @@
+
+        ObjectNode generatedJson = buildManifestFile(settings, fileManifest, dependencies);
+        fileManifest.writeFile(manifestFile.toString(), Node.prettyPrintJson(generatedJson));
+    }
+
+    private static ObjectNode buildManifestFile(
+            GoSettings settings,
+            FileManifest fileManifest,
+            List<SymbolDependency> dependencies
+    ) {
+        Stream<SymbolDependency> nonStdLib = dependencies.stream()
+                .filter(symbolDependency ->
+                        !symbolDependency.getDependencyType().equals(GoDependency.Type.STANDARD_LIBRARY.toString()));
+
+        Optional<SymbolDependency> minStandard = dependencies.stream()
+                .filter(symbolDependency ->
+                        symbolDependency.getDependencyType().equals(GoDependency.Type.STANDARD_LIBRARY.toString()))
+                .max(Comparator.comparing(SymbolDependency::getVersion));
+
+        Map<StringNode, Node> manifestNodes = new HashMap<>();
+
+        Map<String, String> minimumDependencies = gatherMinimumDependencies(nonStdLib);
+
+        Map<StringNode, Node> dependencyNodes = new HashMap<>();
+        for (Map.Entry<String, String> entry : minimumDependencies.entrySet()) {
+            dependencyNodes.put(StringNode.from(entry.getKey()),
+                    StringNode.from(entry.getValue()));
+        }
+
+        Collection<String> generatedFiles = new ArrayList<>();
+        Path baseDir = fileManifest.getBaseDir();
+        for (Path filePath : fileManifest.getFiles()) {
+            generatedFiles.add(baseDir.relativize(filePath).toString());
+        }
+
+        manifestNodes.put(StringNode.from("module"), StringNode.from(settings.getModuleName()));
+        minStandard.ifPresent(symbolDependency ->
+                manifestNodes.put(StringNode.from("go"), StringNode.from(symbolDependency.getVersion())));
+        manifestNodes.put(StringNode.from("dependencies"), Node.objectNode(dependencyNodes));
+        manifestNodes.put(StringNode.from("files"), ArrayNode.fromStrings(generatedFiles));
+
+        return ObjectNode.objectNode(manifestNodes);
+    }
+
+    private static Map<String, String> gatherMinimumDependencies(
+            Stream<SymbolDependency> symbolStream
+    ) {
+        return SymbolDependency.gatherDependencies(symbolStream, (dx, dy) -> {
+            SemanticVersion sx = SemanticVersion.parseVersion(dx.getVersion());
+            SemanticVersion sy = SemanticVersion.parseVersion(dx.getVersion());
+
+            // This *shouldn't* happen in Go since the Go module import path must end with the major version component.
+            // Exception is the case where the major version is 0 or 1.
+            if (sx.getMajor() != sy.getMajor() && !(sx.getMajor() == 0 || sy.getMajor() == 0)) {
+                throw new CodegenException(String.format("Dependency %s has conflicting major versions",
+                        dx.getPackageName()));
+            }
```

Comment:
Comment calls out `Exception is the case where major version is 0 or 1`. 

Do we need an extra condition - where we check for major version as 1 (similar to how we have for major version as 0).

Reply:
I think the idea is to allow a library dependency to be promoted form v0.x to v1.x if something else in the codegen depends on the v1.x version. Also to call out that There is no way to include both v0.x and v1.x of a module in the same application.